common/config/config.go

```diff
@@ -321,6 +321,15 @@ type (
 		RPCName string `yaml:"rpcName"`
 		// Address indicate the remote service address(Host:Port). Host can be DNS name.
 		RPCAddress string `yaml:"rpcAddress"`
+		// AuthorizationProvider contains the information to authorize the cluster
+		AuthorizationProvider struct {
+			// Enable indicates if the auth provider is enabled
+			Enable bool `yaml:"enable"`
+			// Type auth provider type
+			Type string `yaml:"type"` // only supports OAuthAuthorization
+			// PrivateKey is the private key path
+			PrivateKey string `yaml:"privateKey"`
```

Comment:
If this config is per cluster, does that mean they would have to write this per cluster? Is this the most efficient way @longquanzheng ? (Not sure how bigs clusters can be)

Reply:
I never see we do in this way. 
Can you move the struct outside to be independent? 

So that:
```golang
	ClusterInformation struct {
		Enabled                bool  `yaml:"enabled"`
		InitialFailoverVersion int64 `yaml:"initialFailoverVersion"`
		// RPCName indicate the remote service name
		RPCName string `yaml:"rpcName"`
		// Address indicate the remote service address(Host:Port). Host can be DNS name.
		RPCAddress string `yaml:"rpcAddress"`

		AuthorizationProvider AuthorizationProvider `yaml:"authorizationProvider"`
...
	}
		AuthorizationProvider struct {
			// Enable indicates if the auth provider is enabled
			Enable bool `yaml:"enable"`
			// Type auth provider type
			Type string `yaml:"type"` // only supports OAuthAuthorization
			// PrivateKey is the private key path
			PrivateKey string `yaml:"privateKey"`
		}
```